### ISSUE TITLE
grpc to json transcoding issue

### ISSUE THREAD

**Issue by alexshv** [envoyproxy/envoy#6428]
Hi guys. Here's the situation I have: 
Lets assume that I have GRPC service thats talks to REST service. Each service is dockerized Node.js service and has its own envoy instance that talks to other service by envoy to envoy scheme. Now I'm using gRPC-JSON transcoder to transcode GRPC requests to JSON requests. The problem is that REST API endpoint is getting called (it matches POST package/services/method pattern), but the request body is empty. Also when response arrives to the client envoy I'm getting "UNKNOWN: No status received" error on GRPC service. I cannot explicitly set zero status in REST service because Node HTTP module throws.

Here's envoy config file that I have:
```
admin:
  access_log_path: /tmp/admin_access.log
  address:
    socket_address: { address: 0.0.0.0, port_value: 9901 }

static_resources:
  listeners:
  - name: listener_0
    address:
      socket_address: { address: 0.0.0.0, port_value: 51051 }
    filter_chains:
    - filters:
      - name: envoy.http_connection_manager
        typed_config:
          "@type": type.googleapis.com/envoy.config.filter.network.http_connection_manager.v2.HttpConnectionManager
          stat_prefix: grpc_json
          codec_type: AUTO
          route_config:
            name: local_route
            virtual_hosts:
            - name: local_service
              domains: ["*"]
              routes:
              - match:
                  prefix: "/services.Quotes"
                route:
                  cluster: http-proxy
              - match:
                  prefix: "/services.Affiliate"
                route:
                  cluster: affiliate-proxy
          http_filters:
          - name: envoy.grpc_json_transcoder
            config:
              proto_descriptor: "/tmp/envoy/proto.pb"
              services: ["services.Quotes"]
              print_options:
                add_whitespace: true
                always_print_primitive_fields: true
                always_print_enums_as_ints: false
                preserve_proto_field_names: false
          - name: envoy.router
  clusters:
  - name: affiliate-proxy
    connect_timeout: 0.25s
    type: strict_dns
    lb_policy: ROUND_ROBIN
    load_assignment:
      cluster_name: affiliate-proxy
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: affiliate
                port_value: 50061
  - name: http-proxy
    connect_timeout: 0.25s
    type: strict_dns
    lb_policy: ROUND_ROBIN
    load_assignment:
      cluster_name: http-proxy
      endpoints:
      - lb_endpoints:
        - endpoint:
            address:
              socket_address:
                address: http-server
                port_value: 8081
```

Bottom line: I need help in converting protobuf to json and vice versa.

**Comment by mattklein123** [comment]
@lizan

**Comment by lizan** [comment]
Can you also put your proto definition (only the part with HTTP annotation if the proto is too big or you can't expose them)?

**Comment by Kupstas** [comment]
@lizan This is out test application
https://github.com/Kupstas/grpc_json_transcoder. 

I hope it will help

**Comment by lizan** [comment]
Are you trying to using gRPC client to talk to a REST server? If so that is not supported, the grpc_json_transcoder filter is for a REST client talking to a gRPC server.

**Comment by alexshv** [comment]
Understood, thanks. The problem we are trying to solve is that PHP currently does not support GRPC servers. So first thing that comes to mind is to use proxy written in language that does fully support GRPC, it lays in between GRPC service and REST service. Though, Envoy positions itself as a full-feature proxy, so do you consider to implement it in future? Can we open a feature request? Thanks.

**Comment by lizan** [comment]
Another option would be the gRPC reverse bridge filter: https://www.envoyproxy.io/docs/envoy/latest/configuration/http_filters/grpc_http1_reverse_bridge_filter this allows you to use gRPC client to talk to HTTP server, though it doesn't do transcode, so the request and response body have to be protobuf.
